src/adapter/src/coord/sequencer.rs

```diff
@@ -2404,11 +2400,16 @@ impl<S: Append + 'static> Coordinator<S> {
             ExplainStageOld::Timestamp => {
                 let decorrelated_plan = decorrelate(&mut timings, raw_plan)?;
                 let optimized_plan = self.view_optimizer.optimize(decorrelated_plan)?;
-                self.validate_timeline(optimized_plan.depends_on())?;
+                let timeline = self.validate_timeline(optimized_plan.depends_on())?;
                 let source_ids = optimized_plan.depends_on();
-                let id_bundle = self
-                    .index_oracle(compute_instance)
-                    .sufficient_collections(&source_ids);
+                // Determine a timestamp that will be valid for anything in any schema
+                // referenced by the query.
+                let id_bundle = self.timedomain_for(
```

Comment:
Why did this change need to happen? Adding all these IDs to the explain timestamp output seems bad. Won't this make something like `explain timestamp for select * from t` also print all other objects in t's schema?

Reply:
Yes, but that's how timestamps are determined now so it'd be inaccurate to leave them out. When choosing a timestamp for a query we use all IDs in the time-domain in all isolation levels.

Reply:
Ah so if we did leave this as is, then it's just misleading? Hmm can we teach explain timestamp to special case all the other IDs into a single thing that's timeline information or something? I really think we should keep the output of this trimmed down to print things referenced by the query.

Reply:
I'd almost prefer leaving this as is and fixing it in another pr.

Reply:
I think misleading isn't strong enough. Without this change the output is wrong for serializable. Image you have two objects in the same time domain with the following `[since, upper)` ranges: A: `[10, 20)` and B: `[13, 25)`. The timestamp selected to read B under Serializable will be `19`. However, the old explain timestamp will show `24`.

What we could do is add two cases, one for Strict Serializable and one for Serializable. The Strict Serializable one doesn't pull in the whole time domain while the Serializable one does pull in the whole time domain. That way it's still clean for Strict Serializable but also accurate for Serializable.